ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/avdtestbase.py

```diff
@@ -43,6 +46,35 @@ def render(self) -> dict:
         """
         return getattr(self, "test_definition", None) or {}
 
+    def get_host_struct_cfg(self, host: str) -> dict:
+        """
+        Retrieves and returns the structured configuration for a specified host.
+
+        The function fetches the structured_config from hostvars. It ensures the returned object is a
+        dictionary, converting Ansible dictionary-like objects to a standard dictionary if necessary.
+
+        Args:
+            host (str): Hostname to retrieve the structured_config.
+
+        Returns:
+            dict: Structured configuration for the host.
+
+        Raises:
+            AristaAvdError: If host is not in hostvars or if its structured_config is not a dictionary or is empty.
+        """
+        if host not in self.hostvars:
+            raise AristaAvdError(f"Host '{host}' is missing from the hostvars.")
+        struct_cfg = get(self.hostvars, host, separator="..")
+
+        # Check if struct_cfg is a dict or behaves like a dict (e.g. Ansible 'hostvars' object)
+        if not isinstance(struct_cfg, (dict, Mapping)):
+            raise AristaAvdError(f"Host '{host}' structured_config is not a dictionary or dictionary-like object.")
+
+        if not isinstance(struct_cfg, dict):
+            struct_cfg = dict(struct_cfg)
+
+        return struct_cfg
+
     def log_skip_message(
```

Comment:
Line number 82: 
Logging function that logs the test being skipped and appended to a formatted message based on the provided parameters.

Reply:
Fixed in da322e1ab1e922b65e6353b1ab25e092f38d9bf3